Sources/CacheAdvance/CacheReader.swift

```diff
@@ -44,29 +41,26 @@ final class CacheReader {
         reader.offsetInFile
     }
 
-    /// Returns the next encodable message, seeking to the beginning of the next message.
-    func nextEncodedMessage() throws -> Data? {
-        let startingOffset = offsetInFile
-
-        guard startingOffset != offsetInFileAtEndOfNewestMessage else {
-            // We're at the last message.
-            return nil
+    /// Returns the encodable messages in a range
+    func encodedMessagesFromOffset(_ startOffset: UInt64, endOffset: UInt64? = nil) throws -> [Data] {
```

Comment:
I think we can make `endOffset` required given my previous comment.

I'd also consider renaming `endOffset` here to be `maximumOffset` or something to make it more clear that it is the largest possible offset for the end of this message, but that the message may (or even likely!) end before that offset.

Reply:
I comment about this at line: https://github.com/dfed/CacheAdvance/pull/72/files#r1013767862